src/assets/Generator.Shared/Utf8JsonWriterExtensions.cs

```diff
@@ -140,5 +141,53 @@ public static void WriteObjectValue(this Utf8JsonWriter writer, object value)
                     throw new NotSupportedException("Not supported type " + value.GetType());
             }
         }
+
+        public static void WriteBinaryDataValue(this Utf8JsonWriter writer, BinaryData value)
+        {
+            byte[] inputValue = value.ToArray();
+
+            // We will remove the trailing occurrences of the specified characters including: TAB, LF, space.
+            byte[] trimBytes = { 9, 10, 32 };
+            int end = inputValue.Length - 1;
+            for (; end >= 0; end--)
+            {
+                int i = 0;
+                for (; i < trimBytes.Length; i++)
+                {
+                    if (trimBytes[i] == inputValue[end])
+                        break;
+                }
+                if (i == trimBytes.Length)
+                    break;
+            }
+
+            // Check if it is possible to serialize the value as json value.
+            // Possible match kinds: Json Object, Json Array, Json value false, Json value true, Json number.
+            // Our logic here is that, for example, we check if the first and the last bytes are '{' and '}', respectively.
+            if (end >= 0 && (
+                inputValue[0] == 123 && inputValue[end] == 125 ||
```

Comment:
can we add constants so we know what the ints stand for?

Reply:
changed